Add Textlocal Channel

msonowal opened the pull request:
Which provides the ability to send SMS via textlocal.in API

msonowal committed:
Add Textlocal Channel

Which provides the ability to send SMS via textlocal.in API

msonowal committed:
Exception handling

msonowal committed:
Apply Patch from CI

msonowal committed:
modified default case namespace

msonowal committed:
updated exception import which was incorrect previously

msonowal committed:
apply fixes from styleci

msonowal committed:
Update composer.json

msonowal committed:
Update README.md

msonowal committed:
Update TextlocalChannel.php

msonowal committed:
Update README.md

msonowal committed:
Update TextlocalChannel.php

msonowal committed:
Update TextlocalChannel.php

msonowal committed:
Update CouldNotSendNotification.php

msonowal committed:
Added Multiple region textlocal config

msonowal committed:
moved config dir

msonowal committed:
updated contributors name

msonowal committed:
Fixed a bug which occurs when textlocal config is not published and hence no URL is set so resulting in unable to connect to host
Now it merges the configurations with default fallback
added notification channel for on demand support

msonowal committed:
Update TextlocalChannel.php

nikugogoi committed:
auto package discovery for newer laravel versions

need to check if working

msonowal committed:
Merge pull request #2 from nikugogoi/patch-1

auto package discovery for newer laravel versions

msonowal committed:
updated to use api_key from config

updated api_key in constructor which was previously only taking hash

msonowal committed:
Update README.md

msonowal committed:
Update TextlocalServiceProvider.php

atymic commented:
Hi!

Are you still open to creating and maintaining this channel?

If so, could you update it to support laravel 5.5-6.0 please?

Thanks!

msonowal committed:
Added Laravel 6 support

msonowal committed:
Update README.md

msonowal commented:
@atymic 
Updated with laravel 6 support

atymic reviewed (changes requested):
Looks pretty good :)
Please see my comments

msonowal committed:
Update composer.json

Co-Authored-By: atymic <50683531+atymic@users.noreply.github.com>

msonowal committed:
Update composer.json

Co-Authored-By: atymic <50683531+atymic@users.noreply.github.com>

msonowal committed:
Update phpunit.xml.dist

Co-Authored-By: atymic <50683531+atymic@users.noreply.github.com>

atymic commented:
When you have a chance could you have a look at the feedback I added please :)

ShankarAkunuri committed:
Adding unicode support to SMS. (#8)

* Update Textlocal.php

Adding unicode SMS support.

* Update TextlocalChannel.php

Adding unicode SMS support.

msonowal committed:
updated minimum php  version to 7.0

msonowal committed:
Update TextlocalChannel.php

msonowal committed:
Update travis.yml

msonowal commented on src/TextlocalServiceProvider.php:
```diff
@@ -0,0 +1,51 @@
+                function () {
+                    $textlocalConfig = config('services.sms.textlocal');
+
+                    return new Textlocal(
```
@atymic 
As per laravel it maintains external service creds are manage in `services.php` so for that it is like that.
Anyway if you want me to set it in own configuration let me know

msonowal replied:
@atymic 
I have updated

msonowal committed:
Update textlocal.php

msonowal committed:
Update TextlocalServiceProvider.php

msonowal committed:
Update Textlocal.php

msonowal commented on src/Textlocal.php:
```diff
@@ -0,0 +1,868 @@
+ * @const      REQUEST_TIMEOUT   Timeout in seconds for the HTTP request
+ * @const      REQUEST_HANDLER   Handler to use when making the HTTP request (for future use)
+ */
+class Textlocal
```
@atymic  No we get a zip sdk if you email to textlocal for core file

atymic replied:
Oh ok.

msonowal committed:
Update Textlocal.php

msonowal committed:
Update composer.json

msonowal committed:
Update TextlocalChannel.php

msonowal committed:
Update TextlocalChannel.php

msonowal committed:
Update TextlocalChannel.php

msonowal committed:
Update TextlocalChannel.php

msonowal commented on src/TextlocalChannel.php:
```diff
@@ -0,0 +1,61 @@
+        //TODO check if numbers are correct
+
+        // Get the message from the notification class
+        $message = (string) $notification->toSms($notifiable);
```
@atymic Can you tell more about this

atymic replied:
Should use the service name

msonowal commented on tests/ExampleTest.php:
```diff
@@ -0,0 +1,12 @@
+class ExampleTest extends \PHPUnit_Framework_TestCase
+{
+    /** @test */
+    public function true_is_true()
```
@atymic 
Can we add this later

atymic replied:
Please add them before we create the repo

msonowal commented:
> When you have a chance could you have a look at the feedback I added please :)

I have updated the codes with the changes requested let me know

msonowal committed:
updated method routeNotificationFor('sms') to routeNotificationFor('Textlocal')

updated method routeNotificationFor('sms') to routeNotificationFor('Textlocal')
instead of generic

msonowal committed:
Update textlocal.php

msonowal committed:
Update README.md

msonowal committed:
Update README.md

msonowal commented:
ping @atymic

atymic commented on README.md:
```diff
@@ -1,5 +1,107 @@
+This package allows to send SMS using Textlocal API using laravel notifications
 
-Discuss about new channel proposals our share your finished channels in the [proposal issue](https://github.com/laravel-notification-channels/new-channels/issues/6).
+Supports Laravel 5.x to 6.x
```
```suggestion
Supports Laravel 5.5 to 6.x
```

atymic commented on README.md:
```diff
@@ -1,5 +1,107 @@
+Supports Laravel 5.x to 6.x
 
-Take a look at our [FAQ](http://laravel-notification-channels.com/) to see our small list of rules, to provide top-notch notification channels.
+# Found any bugs or improvement open an issue or send me a PR
```
This should be the channel title

atymic commented on README.md:
```diff
@@ -1,5 +1,107 @@
+[![License](https://poser.pugx.org/msonowal/laravel-notification-channel-textlocal/license)](https://packagist.org/packages/msonowal/laravel-notification-channel-textlocal)
+[![Total Downloads](https://poser.pugx.org/msonowal/laravel-notification-channel-textlocal/downloads)](https://packagist.org/packages/msonowal/laravel-notification-channel-textlocal)
+
+This package makes it easy to send notifications using [textlocal](https://www.textlocal.in/) with Laravel 5.3.+
```
```suggestion
This package makes it easy to send notifications using [textlocal](https://www.textlocal.in/) with Laravel 5.5+ & 6.x
```

atymic commented on README.md:
```diff
@@ -1,5 +1,107 @@
+
+### Available Message methods
+
+A list of all available options
```
This is missing

atymic commented on composer.json:
```diff
@@ -0,0 +1,47 @@
+{
+    "name": "msonowal/laravel-notification-channel-textlocal",
+    "description": "This Driver for textlocal sms to send SMS via Textlocal API through laravel notification",
```
```suggestion
    "description": "Textlocal Notifications Channel for Laravel",
```

atymic commented on phpunit.xml.dist:
```diff
@@ -0,0 +1,29 @@
+         processIsolation="false"
+         stopOnFailure="false">
+    <testsuites>
+        <testsuite name=":service_name Test Suite">
```
Still missing

msonowal committed:
Update composer.json

Co-Authored-By: atymic <atymicq@gmail.com>

msonowal committed:
Update README.md

Co-Authored-By: atymic <atymicq@gmail.com>

tsainadh committed:
Update TextlocalChannel.php (#12)

set sender id from notification if having multiple sender id's to manage.

atymic commented:
@msonowal Are you able to make the requested changes?

atymic commented:
@msonowal 

Did you get a chance to review above?

msonowal commented:
Hi @atymic 
Currently I am busy, I have this in mind will do when I get some time.
maybe within next month

atymic commented:
No worries :)

Saneesh commented:
Hello @msonowal and @atymic 
When Textlocal channel will support L6/7?

Another question: how can I join to contribute like you guys are doing in this repository?

msonowal commented:
Send a PR, I am really busy  atm

msonowal commented:
@atymic 
Hi can you check the 
https://github.com/msonowal/laravel-notification-channel-textlocal/tree/wip-tests-andlaravel7-support

branch `wip-tests-andlaravel7-support`

and also help to fix the test as I have updated the test skelton to test but not able to fix it

msonowal commented:
> Hello @msonowal and @atymic
> When Textlocal channel will support L6/7?
> 
> Another question: how can I join to contribute like you guys are doing in this repository?

You can use the branch 
`wip-tests-andlaravel7-support`

it has 6 and 7 support

atymic commented:
@msonowal 

Please PR code here:
https://github.com/laravel-notification-channels/textlocal

And i'll get everything set up for release :)

atymic commented:
Launch checklist https://github.com/laravel-notification-channels/textlocal/issues/1

msonowal commented:
Sure thing

atymic commented:
Can you PR on the new repo please?
Otherwise will close this and archive shortly.

msonowal commented:
@atymic 

PR https://github.com/laravel-notification-channels/textlocal/pull/3

dependabot-preview[bot] committed:
Upgrade to GitHub-native Dependabot (#13)

Co-authored-by: dependabot-preview[bot] <27856297+dependabot-preview[bot]@users.noreply.github.com>

rupamjbordoloi committed:
Laravel 7.x & 8.x support (#14)

* laravel 8 support

* readme updated

Co-authored-by: rupam4 <rupam@claritytech.io>

msonowal committed:
updated support for Laravel 9 LTS

msonowal committed:
Update README.md

msonowal committed:
Update README.md

msonowal committed:
Update README.md

msonowal committed:
Update README.md

msonowal committed:
Update README.md

msonowal committed:
Update README.md

Update README.md

msonowal committed:
feat: custom client configuration usage via per notification or  notifiable

Update Textlocal.php, TextlocalChannel.php, TextlocalServiceProvider.php

Create INotificationUsesTextlocalClientConfig.php

Create IUsesTextlocalClientConfig.php

Update README.md

msonowal committed:
Update TextlocalChannel.php

Update README.md

Update CHANGELOG.md

msonowal committed:
Update INotificationUsesTextlocalClientConfig.php

Update IUsesTextlocalClientConfig.php

msonowal committed:
Update TextlocalChannel.php

msonowal committed:
Update textlocal.php config with sample/example stub for custom usage

msonowal committed:
feat: added support for laravel 10 Update composer.json

Update README.md

lrljoe committed:
Add Laravel 11 Support
